### ISSUE TITLE
[GlobalStep] Stock count fails to update for "on backorder" products under Task Managment Area on "WooCommerce->Home" Page Turn on screen reader support

### ISSUE THREAD

**Issue by gglobalstep** [woocommerce/woocommerce-admin#6845]
## Bug Description:
Stock count fails to update for "on backorder" products under Task Management Area on "WooCommerce->Home" Page

## Test Instructions: Task Management Area
https://github.com/woocommerce/woocommerce-admin/wiki/Smoke-Test-Checklist#task-managment-area

## Environment:
Woocommerce Version : WooCommerce 5.2.2
WooCommerce Admin: v2.3.0 Beta 1 and v2.1.5
WordPress version: (v5.7)

**PC:**
Windows 10, Mac 10.14.6
Chrome(Version 89.0.4389.90) 
Firefox(Version 87.0)
Safari: v13.1.1

## Steps To Reproduce:
1. Create any test site using JN site.
1. Install and activate all the required plugins 
1. Install and activate WooCommerce Payments plugin.
1. Complete the onboarding setup wizard.
1. Complete the Task List and tap on three dots for this section and click on "Hide this" option
1. Go to "Products->Add New" and Create a Product say T-shirt
1. Go to product "T-shirt" Inventory and select "Manage stock?"
1. Set the stock quantity "zero" and "Allow" backorders
1. Assuming the low threshold for stock is set to 2 by default
1. Click on "Publish" button
1. Go to WooCommerce->Home
1. Observe that Stock count fails to update for "on backorder" products under Task Management Area on "WooCommerce->Home" Page

## Actual Result:
Stock count fails to update for "on backorder" products under Task Management Area on "WooCommerce->Home" Page

## Expected Result:
Stock count should update for "on backorder" products under Task Management Area on "WooCommerce->Home" Page

## Video:

https://user-images.githubusercontent.com/41110392/115590835-0afa6e80-a2ef-11eb-9805-d6f32b41b4ad.mp4

**Isolating the problem (mark completed items with an [x]):**
- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

<details>
```
 `
### WordPress Environment ###

WordPress address (URL): https://awkward-flyingfish.jurassic.ninja
Site address (URL): https://awkward-flyingfish.jurassic.ninja
WC Version: 5.2.2
REST API Version: ✔ 5.2.2
WC Blocks Version: ✔ 4.7.2
Action Scheduler Version: ✔ 3.1.6
WC Admin Version: ✔ 2.3.0-beta.1
Log Directory Writable: ✔
WP Version: 5.7.1
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: ✔
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: Apache/2.4.46 (Unix) OpenSSL/1.0.2g
PHP Version: 7.4.16
PHP Post Max Size: 1 GB
PHP Time Limit: 30
PHP Max Input Vars: 5000
cURL Version: 7.47.0
OpenSSL/1.0.2g

SUHOSIN Installed: –
MySQL Version: 5.7.33-0ubuntu0.16.04.1-log
Max Upload Size: 512 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 5.2.2
WC Database Prefix: wp_
Total Database Size: 5.54MB
Database Data Size: 4.04MB
Database Index Size: 1.50MB
wp_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Data: 0.02MB + Index: 0.11MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_options: Data: 3.05MB + Index: 0.09MB + Engine InnoDB
wp_postmeta: Data: 0.13MB + Index: 0.09MB + Engine InnoDB
wp_posts: Data: 0.06MB + Index: 0.06MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB

### Post Type Counts ###

attachment: 28
customize_changeset: 1
nav_menu_item: 7
page: 7
post: 2
product: 20
product_variation: 7
revision: 1
shop_order: 2

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (11) ###

Query Monitor: by John Blackbourn – 3.6.7
Companion Plugin: by Osk – 1.18
Facebook for WooCommerce: by Facebook – 2.3.5
Gutenberg: by Gutenberg Team – 10.4.1
Jetpack by WordPress.com: by Automattic – 9.6.1
Google Ads & Marketing by Kliken: by Kliken – 1.0.7
WooCommerce Admin: by WooCommerce – 2.3.0-beta.1
WooCommerce Payments: by Automattic – 2.2.0
WooCommerce Shipping & Tax: by Automattic – 1.25.11
WooCommerce: by Automattic – 5.2.2
WP Crontrol: by John Blackbourn & crontributors – 1.10.0

### Inactive Plugins (2) ###

Akismet Anti-Spam: by Automattic – 4.1.9
Hello Dolly: by Matt Mullenweg – 1.7.2

### Dropin Plugins (1) ###

db.php: Query Monitor Database Class

### Settings ###

API Enabled: –
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –

### WC Pages ###

Shop base: #6 - /shop/
Cart: #7 - /cart/
Checkout: #8 - /checkout/
My account: #9 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Twenty Twenty-One
Version: 1.3
Author URL: https://wordpress.org/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Action Scheduler ###

Complete: 8
Oldest: 2021-04-21 09:33:14 +0000
Newest: 2021-04-21 10:02:03 +0000


### Status report information ###

Generated at: 2021-04-21 10:42:41 +00:00
`
```
</details>

**Comment by timmyc** [comment]
Thank you for the detailed defect report - I can confirm this using the steps outlined above. It looks like the reason the stock panel is not showing in this exact scenario is that [we have some logic](https://github.com/woocommerce/woocommerce-admin/blob/main/client/homescreen/activity-panel/panels.js#L43-L45) that does not show the stock panel if there are no orders placed on a store. Once I added a completed order, the panel appears with the backordered product as expected.

Given this code was added in https://github.com/woocommerce/woocommerce-admin/pull/5973 as per the spec, I don't think this is a bug, rather the test instructions just did not properly specify the need for an order to have been completed.
